src/imu_diag_data_provider.cpp

```diff
@@ -0,0 +1,59 @@
+/***************************************************************************//**
+*   @file   imu_diag_data_provider.cpp
+*   @brief  Implementation for imu diag data
+*   @author Vasile Holonec (Vasile.Holonec@analog.com)
+********************************************************************************
+* Copyright 2023(c) Analog Devices, Inc.
+
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "imu_ros2/imu_diag_data_provider.h"
+
+ImuDiagDataProvider::ImuDiagDataProvider()
+{
+    init();
+}
+
+ImuDiagDataProvider::~ImuDiagDataProvider()
+{
+
+}
+
+void ImuDiagDataProvider::init()
+{
+    // initialize a library
+}
+
+imu_ros2::msg::ImuDiagData ImuDiagDataProvider::getData(int count)
+{
+    (int)count;
```

Comment:
What is the usage of this count parameter? It does not look like this is used anywhere in the code...or am I mistaken? Does it have a purpose? If not, let's get rid of it.